client/src/components/JurisdictionAdmin/DownloadBatchTallySheetsButton.tsx

```diff
@@ -23,15 +24,16 @@ const DownloadBatchTallySheetsButton = ({
   const batchesQuery = useBatches(electionId, jurisdictionId, roundId)
   const contests = useContestsJurisdictionAdmin(electionId, jurisdictionId)
 
-  if (!batchesQuery.isSuccess || !contests) {
-    return null
-  }
-
-  const { batches } = batchesQuery.data
-  // Batch comparison audits only support a single contest
-  const contest = contests[0]
-
   const onClick = async () => {
+    // Wait for the batches/contests to load in case they haven't yet.
+    while (!batchesQuery.isSuccess || contests === null) {
+      // eslint-disable-next-line no-await-in-loop
+      await sleep(100)
+    }
```

Comment:
That's a fair concern. Maybe I should make it wait for `batchesQuery.isSettled`, which will account for both request success and failure? I could also add a timeout.

I want to push for the best UX if possible - an unexplained disabled button feels a bit confusing. We could also just load the data on button press instead of in advance. What do you think? I know we discussed this a bit once when you originally implemented this button.

Reply:
Agreed that the disabled button isn't the best UX!

Waiting for `batchesQuery.isSettled` (and the relevant analog for `contests`) sounds great! Basically just want to make sure that we break out of the loop even under error circumstances

My gut tells me that a timeout isn't may not be worth the complexity but don't feel strongly

Reply:
Ok turns out `isSettled` is only available in a newer version of react-query, but I was able to still handle the error case (ebd801cf). There isn't an equivalent error case for contests, so I converted that to use react-query (fb0e2725).